### PR TITLE
SQL: Fix issue with LIKE/RLIKE as painless script

### LINKED ISSUE #53486: SQL: LIKE/RLIKE don't produce painless scripts
LIKE/RLIKE don't produce painless scripts and wrong results are returned, e.g.:
```
SELECT count(*), CASE WHEN text.keyword LIKE '%a%' then 1 ELSE 2 END as c FROM test_termvector GROUP BY t
```
will return one group with value 2.

### PR DESCRIPTION
**Pull request by matriv**
Add missing asScript() implementation for LIKE/RLIKE expressions.

When LIKE/RLIKE are used for example in GROUP BY or are wrapped with
scalar functions in a WHERE clause, the translation must produce a
painless script which will be executed to implement the correct
behaviour and previously this was completely missing, and as a
consquence wrong results were silently (no error) returned.

Fixes: #53486

### CODE REVIEW

**Comment by matriv** [comment]
I decided on a `Pattern` iface because I didn't like to use `LikePattern#toString() => asJavaRegex`, I felt it could be confusing, but on the other hand it kind of feels a bit "too much" to add that iface.

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/Pattern.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.ql.expression.predicate.regex;
+
+interface Pattern {
```
Comment (costin) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/Pattern.java]: Maybe `RegexPattern`

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/RegexMatch.java [hunk 2]
```diff
@@ -54,8 +58,30 @@ public boolean foldable() {
...
+        return RegexProcessor.RegexOperation.match(val, pattern().asJavaRegex());
+    }
+
     @Override
+    protected Processor makeProcessor() {
+        return new RegexProcessor(pattern().asJavaRegex());
+    }
+
+    @Override
+    public ScriptTemplate asScript() {
```
Comment (costin) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/RegexMatch.java]: 👍

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/Pattern.java (hunk 1 above)
Comment (matriv) [reply x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/Pattern.java]: I'm not sure, the LikePattern is not 100% regex pattern.

Comment (matriv) [reply x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/Pattern.java]: How about `StringMatchPattern`?

Comment (costin) [reply x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/Pattern.java]: Why not `StringPattern` ?

Comment (matriv) [reply x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/Pattern.java]: sure that too!

**Review by bpintea — APPROVED** [review]

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**